lib/pom/Endpoints_page.js

```diff
@@ -0,0 +1,80 @@
+var Page = require('./Page');
+var Button_object = require('ui_test_automation/wrappers/Button_object');
+var Input_object = require('ui_test_automation/wrappers/Input_object');
+var DropDown_object = require('ui_test_automation/wrappers/DropDown_object');
+
+class Endpoints_page extends Page {
+  
+  get OAS_ENDPOINTS_BUTTON() {return new Button_object('a*=Endpoints');}
+  get OAS_ADD_NEW_ENDPOINT_BUTTON() {return new Button_object('span*=Add New Endpoint');}
+  get OAS_ENDPOINT_METHOD_DROPDOWN() {return $('//label[text()="Method"]//following::span[1]');}
+  get OAS_ENDPOINT_PATH_INPUT() {return new Input_object('//input[@placeholder="Type path"]');}
+  get OAS_ADD_ENDPOINT_BUTTON() {return new Button_object('span*=Add Endpoint');}
+  get OAS_ENDPOINT_WHITELIST_BUTTON() {return new Button_object('span*=Add Whitelist');}
+  get OAS_OPTIONS_BUTTON() {return new Button_object('span*=OPTIONS');}
+  get OAS_EDIT_ENDPOINT_BUTTON() {return new Button_object('span*=Edit Endpoint');}
+  get OAS_REMOVE_ENDPOINT_BUTTON() {return new Button_object('//ul[@class="tyk-dropdown-menu tyk-dropdown opened"]//following::span[text()="Remove Endpoint"]');}
+  get OAS_SEARCH_ENDPOINTS_INPUT() {return new Input_object('//input[@placeholder="Search Endpoints"]');}
+  get OAS_FILTER_METHOD_DROPDOWN() {return new DropDown_object('//span[contains(text(), "Show")]//following::span[1]');}
+  get OAS_REMOVE_CONFIRMATION_BUTTON() {return new Button_object('//button/span[text()="Remove Endpoint"]');}
+  get OAS_UPDATE_ENDPOINT_BUTTON() {return new Button_object('span*=Update Endpoint');}
+  
+  clickOnEndpoint(endpoint, method) {
+    let endpointUrlLink = encodeURIComponent(endpoint.split('/').join('-')) + method;
+    let endpointSelector = $(`//a[contains(@href, "${endpointUrlLink}")]`);
+    endpointSelector.click();
```

Comment:
please add any logs in here or use new Button_object(`//a[contains(@href, "${endpointUrlLink}")]`)